libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -852,11 +852,7 @@ - (void)setupWithCredentials:(SFOAuthCredentials*) credentials {
 
 - (void)restartAuthentication {
     @synchronized (self.authBlockList) {
-        if (!self.authenticating) {
-            [self log:SFLogLevelWarning format:@"%@: Authentication manager is not currently authenticating.  No action taken.", NSStringFromSelector(_cmd)];
-            return;
-        }
-        [self log:SFLogLevelInfo format:@"%@: Restarting in-progress authentication process.", NSStringFromSelector(_cmd)];
+        [self log:SFLogLevelInfo format:@"%@: Restarting authentication process.", NSStringFromSelector(_cmd)];
```

Comment:
I'm not seeing in the flow where we consider authentication over if canceled. Where does that cancelation occur?  At least in your embedded VC flow.  And I actually prefer your behavior—not "finishing" authentication in this case—because it otherwise breaks the contract passed in to the public login methods by the consumer.

The restart process was actually implemented to solve these kinds of auth-internal use cases.  I.e. don't "finish" the authentication, but do a soft reset, and then the next iteration can still call the appropriate completion handlers that were configured originally.

So I'd opt for keeping the `self.authenticating` guard.